lib/ecto/postgres/pg_rational_migrations.ex

```diff
@@ -463,6 +486,42 @@ defpgmodule Vtc.Ecto.Postgres.PgRational.Migrations do
     :ok
   end
 
+  @doc section: :migrations_functions
+  @doc """
+  Creates Creates `rational_private.modulo(a, b)` backing function for the `%` operator
+  between two rationals.
+  """
+  @spec create_func_modulo() :: :ok
+  def create_func_modulo do
+    abs = function(:abs, Migration.repo())
+    minus = function(:minus, Migration.repo())
+
+    create_func =
+      "#{private_function_prefix(Migration.repo())}modulo"
+      |> :erlang.binary_to_atom(:utf8)
+      |> Postgres.Utils.create_plpgsql_function(
+        args: [dividend: :rational, divisor: :rational],
+        declares: [
+          dividend_abs: {:rational, "#{abs}(dividend)"},
+          quotient: {:rational, "dividend_abs / divisor"},
+          quotient_floored: {:rational, "(floor((quotient).numerator::float / (quotient).denominator), 1)"},
```

Comment:
is it possible to pass through decimal values for the numerator and denominator? If not, why not use `integer` instead of `float`?

Reply:
Yeah I don't love what I'm doing here. I'm trying to match the modulo behavior of the cpython `fractions` package. I figure that's probably the best reference I have for a "correct" implementation. They rounds towards negative Infinity when doing this operation.

But Postgres rounds towards zero, so I can't use integer division like I am doing in the application code. I have to cast to float, then floor, otherwise we round the wrong way on negative numbers.

I'll take a look at decimal! I also looked up how the Python package implements this and they manage to do it with straight integer math and no divisions, so I'm gonna give their formula a shot too.

Reply:
Alright, did a deep dive on division. I originally implemented [the first vtc](https://github.com/opencinemac/vtc-py) in Python. Python has floored integer division so for division:

```python
>>> 13 // 3
4
>>> -13 // 3
-5
```

and for remainder:

```python
>>> 13 % 3
1
>>> -13 % 3
2
```

This same logic also applies to python's fraction library:

```python
>>> from fractions import Fraction
>>>
>>> Fraction(13, 1) % Fraction(3, 1)
Fraction(1, 1)
>>>
>>> Fraction(13, 1) % Fraction(-3, 1)
Fraction(-2, 1)
```

When I was originally implementing this library, my M.O. was to use the tests from `vtc-py` and re-implement them exactly in `vtc-ex`. Since [Ratio]() -- the library that backs all the rational calculations for vtc -- doesn't ship with a remainder function, when I went to implement one, I copied the round-towards-negative-infinity behavior of the python library.

But in elixir -- and Postgres -- integer division rounds towards zero:

```elixir
iex(1)> rem(13, 3)
1
iex(2)> rem(-13, 3)
-1
```

So I kinda had to fight both languages to use floored integer division like Python naturally does. After looking at this for a bit, I think I actually prefer rounding towards zero. It means that positive and negative operations are symmetrical. It also makes the math a lot simpler, especially in Postgres. 

I think there is something to *every vtc library yields the same results for the same calculations* that I might want to revisit in the future, but maybe there is something *more* to *each vtc implementation follows the mathematical conventions of the platform it is running on*.

I've updated both application and Postgres logic to use truncation for their `rem` and `%` operations and added a bunch of tests. I've also bumped the minor version of the lib to indicate pre-1.0 breaking changes.

Changes in fae5d19ba5d457b42c0ecb3cb3917eee09ff365e.

Reply:
the changes look good to me — the new version results in much simpler code which is always a plus!